code/modules/mob/new_player/new_player.dm

```diff
@@ -146,7 +146,7 @@
 			if(!SSticker || SSticker.current_state == GAME_STATE_STARTUP)
 				to_chat(src, "<span class='warning'>The game is still setting up, please try again later.</span>")
 				return
-			if(alert("Are you sure you wish to observe?\nYou will have to wait at least 5 minutes before being able to respawn!", "Observe", "Yes", "No") == "Yes")
+			if(alert("Are you sure you wish to observe?\nYou will have to wait at least 30 or 2 minutes before being able to respawn as a marine or alien, respectively!", "Observe", "Yes", "No") == "Yes")
```

Comment:
```suggestion
			if(alert("Are you sure you wish to observe?\nYou might have to wait a certain time to respawn or be unable to, depending on the game mode!", "Observe", "Yes", "No") == "Yes")
```

A more generic handling would be better.
It would be even better to do something like this instead:
```
			if(alert("Are you sure you wish to observe?[SSticker.mode?.observe_respawn_message()]", "Observe", "Yes", "No") == "Yes")
```
And then defining the proc, something like this:
```
/datum/game_mode/proc/observe_respawn_message()
	return "\nYou might have to wait a certain time to respawn or be unable to, depending on the game mode!"

/datum/game_mode/infestation/observe_respawn_message()
	return "\nYou will have to wait at least [GLOB.respawntime * 0.1] or [GLOB.xenorespawntime * 0.1] seconds before being able to respawn as a marine or alien, respectively!"
```